### ISSUE TITLE
Can't run lsp-dart in a dart file

### ISSUE THREAD

**Issue by softode-code** [emacs-lsp/lsp-dart#195]
I can not run dart-mode in doom-emacs. I have defined it in lang: as 

(dart +flutter +lsp)

I have copied the config from https://emacs-lsp.github.io/lsp-dart/ exmaple:

```
(require 'package)
(add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/") t)
(package-initialize)

(setq package-selected-packages
  '(dart-mode lsp-mode lsp-dart lsp-treemacs flycheck company
    ;; Optional packages
    lsp-ui company hover))

(when (cl-find-if-not #'package-installed-p package-selected-packages)
  (package-refresh-contents)
  (mapc #'package-install package-selected-packages))

(add-hook 'dart-mode-hook 'lsp)

(setq gc-cons-threshold (* 100 1024 1024)
      read-process-output-max (* 1024 1024))

```

**Screenshots**
![image](https://user-images.githubusercontent.com/61111989/234305197-de90f149-2782-4c0c-a9c9-b13b21d08a53.png)

**Comment by ericdallo** [comment]
Do you have flutter / dart available in your `$PATH`? if so, try starting emacs from terminal which has that available

**Comment by softode-code** [comment]
Yes, I have flutter in my PATH and I started emacs from that terminal. Still the same result.

**Comment by ericdallo** [comment]
Alright, could you follow the issue template and include here the result of: <kbd>M-x</kbd> `lsp-dart-version`?

**Comment by softode-code** [comment]
I get the following result when I run lsp-dart-version:

[LSP Dart] 1.24.1 at 2023.04.26 @ Emacs 28.2
[Dart SDK] 
 [33m ▄      [31m      ▄▄      [32m      ▄▄      [34m      ▄▄      [35m      ▄▄      [36m      ▄▄     
 [33m ███▄▄  [31m   ██▬██▬██   [32m   ██▬██▬██   [34m   ██▬██▬██   [35m   ██▬██▬██   [36m   ██▬██▬██  
 [33m █████▀▀[31m     ████     [32m     ████     [34m     ████     [35m     ████     [36m     ████    
 [33m █▀▀    [31m   ▄██  ██▄   [32m   ▄██  ██▄   [34m   ▄██  ██▄   [35m   ▄██  ██▄   [36m   ▄██  ██▄  
 [33m █      [31m ▄▄▄▀█  █▀▄▄▄ [32m ▄▄▄▀█  █▀▄▄▄ [34m ▄▄▄▀█  █▀▄▄▄ [35m ▄▄▄▀█  █▀▄▄▄ [36m ▄▄▄▀█  █▀▄▄▄
 [33m▄█▄     [31m ███▀████▀███ [32m ███▀████▀███ [34m ███▀████▀███ [35m ███▀████▀███ [36m ███▀████▀███
            [31m  ▀  ▀        [32m  ▀  ▀        [34m  ▀  ▀        [35m  ▀  ▀        [36m  ▀  ▀       
[1m
 [93m ▄      [91m      ▄▄      [92m      ▄▄      [94m      ▄▄      [95m      ▄▄      [96m      ▄▄     
 [93m ███▄▄  [91m   ██▬██▬██   [92m   ██▬██▬██   [94m   ██▬██▬██   [95m   ██▬██▬██   [96m   ██▬██▬██  
 [93m █████▀▀[91m     ████     [92m     ████     [94m     ████     [95m     ████     [96m     ████    
 [93m █▀▀    [91m   ▄██  ██▄   [92m   ▄██  ██▄   [94m   ▄██  ██▄   [95m   ▄██  ██▄   [96m   ▄██  ██▄  
 [93m █      [91m ▄▄▄▀█  █▀▄▄▄ [92m ▄▄▄▀█  █▀▄▄▄ [94m ▄▄▄▀█  █▀▄▄▄ [95m ▄▄▄▀█  █▀▄▄▄ [96m ▄▄▄▀█  █▀▄▄▄
 [93m▄█▄     [91m ███▀████▀███ [92m ███▀████▀███ [94m ███▀████▀███ [95m ███▀████▀███ [96m ███▀████▀███
            [91m  ▀  ▀        [92m  ▀  ▀        [94m  ▀  ▀        [95m  ▀  ▀        [96m  ▀  ▀       
[0m  
Dart SDK version: 2.19.6 (stable) (Tue Mar 28 13:41:04 2023 +0000) on "linux_x64"

[Flutter SDK] /home/umer/Development/flutter/
[Flutter project] true
[Project entrypoint] /home/umer/Development/quizapp/lib/main.dart

**Comment by softode-code** [comment]
@ericdallo please support on this.

**Comment by ericdallo** [comment]
@softode-code sorry for the delay.

The output looks correct (besides the weird dart sdk chars), and I noticed you are using doom-emacs, so no extra config from https://emacs-lsp.github.io/lsp-dart/ is necessary, you can check my dotfiles for my doom-emacs [as example](https://github.com/ericdallo/dotfiles/blob/master/.config/doom/init.el#L122).

could you try on this [sample project](https://github.com/ericdallo/flutter_sample)?

**Comment by softode-code** [comment]
Still the same result.

**Comment by ericdallo** [comment]
Could you try [lsp-start-pain.el](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/#check-if-any-package-or-wrong-config-could-be-causing-the-issue) ?

**Comment by softode-code** [comment]
I ran "lsp-start-plain", a new emacs window opened with the below messages:

![image](https://github.com/emacs-lsp/lsp-dart/assets/61111989/a38639cc-32fd-4428-aabe-cc684c9ba64b)

**Comment by softode-code** [comment]
Here is the logs from "*lsp-log*":

Command "/home/umer/Development/flutter/bin/cache/dart-sdk/bin/dart language-server --client-id emacs.lsp-dart --client-version 1.24.1" is present on the path.

**Comment by ericdallo** [comment]
That was probably an instability in melpa for downloading packages, trying again should work

**Comment by softode-code** [comment]
I again ran "lsp-start-plain", a new emacs window opened. I opened a dart file in it but same results:

![image](https://github.com/emacs-lsp/lsp-dart/assets/61111989/38255ee3-2f2a-4b88-aee3-3ee8e889c858)
